File: github_code/insertion-sort/cpp/113.c
// insertionsort.cpp

#include <iostream>
#include <vector>
#include <cstdlib>

using namespace std;

// insertion sort
void ins_sort(vector<int> & v)
{
	int i, j, elem;
	int n = v.size();

	for (i = 1; i < n; i++)
   {	
		for (elem = v[i], j = i-1; j >=0 and elem < v[j]; j--)
   	{
			v[j+1] = v[j];
		}
		v[j+1] = elem;
	}
}

// main program
int main()
{
	srand(time(0)); // initialize random number generator
	int s; // initialize input size

	cin >> s;

	vector<int> v(s); // initialize vector with user input size
	
	// randomize numbers
	for(int i=0; i < v.size(); ++i)
	{
		v[i] = rand() % 1000001;
	}

	ins_sort(v); // bubble sort;

	for(int i=0; i < v.size(); ++i)
	{
		cout << v[i] << endl;
	}
}

File: github_java/4/113.java
package SortingProgrammms;

public class insertionsort {
	
	public class MyInsertionSort {
		 
	    public  void main(String[] args) {
	         
	        int[] input = { 4, 2, 9, 6, 23, 12, 34, 0, 1 };
	        insertionSort(input);
	    }
	     
	    private  void printNumbers(int[] input) {
	         
	        for (int i = 0; i < input.length; i++) {
	            System.out.print(input[i] + ", ");
	        }
	        System.out.println("\n");
	    }
	 
	    public  void insertionSort(int array[]) {
	        int n = array.length;
	        for (int j = 1; j < n; j++) {
	            int key = array[j];
	            int i = j-1;
	            while ( (i > -1) && ( array [i] > key ) ) {
	                array [i+1] = array [i];
	                i--;
	            }
	            array[i+1] = key;
	            printNumbers(array);
	        }
	    }
	

	}}
